diff.go

```diff
@@ -113,6 +115,26 @@ func (d *differ) diffApplication(name string) *ApplicationDiff {
 	effectiveBundleExpose := bundle.Expose || len(bundle.ExposedEndpoints) != 0
 	effectiveModelExpose := model.Exposed || len(model.ExposedEndpoints) != 0
 
+	// If any of the sides is exposed but lacks any expose endpoint details
+	// assume an implicit expose "" to 0.0.0.0/0 for comparison purposes.
```

Comment:
Can we still make this assumption?

Reply:
I believe this is needed this for backwards compatibility when calculating diffs with pre 2.9 bundles that only specify `expose: true`. 

Note that the migration step in the linked PR will inject 0.0.0.0/0 and ::/0 for the wildcard endpoint for any exposed app. Having this extra logic here allows us to correctly perform the diff regardless of which controller version provided the current model state to diff against. 

It's kind of an edge case but it allows us to produce consistent diffs.

Reply:
Also, there is a followup PR (https://github.com/juju/bundlechanges/pull/71) that updates this to also include the IPV6 CIDRs